sdk/python/feast/templates/cassandra/bootstrap.py

```diff
@@ -0,0 +1,251 @@
+        'c_protocol_version'
+        'c_local_dc'
+        'c_load_balancing_policy'
+    """
+    write_setting_or_remove(
+        config_file,
+        settings["c_secure_bundle_path"],
+        "secure_bundle_path",
+        "/path/to/secure/bundle.zip",
+    )
+    #
+    if settings["c_hosts"]:
+        replace_str_in_file(
+            config_file,
+            "        - 127.0.0.1",
+            os.linesep.join(f"        - {c_host}" for c_host in settings["c_hosts"]),
+        )
+    else:
+        remove_lines_from_file(config_file, "hosts:")
+        remove_lines_from_file(config_file, "- 127.0.0.1")
+    #
+    write_setting_or_remove(
+        config_file, settings["c_port"], "port", "9042",
+    )
+    #
+    write_setting_or_remove(
+        config_file, settings["c_username"], "username", "c_username",
+    )
+    #
+    write_setting_or_remove(
+        config_file, settings["c_password"], "password", "c_password",
+    )
+    #
+    replace_str_in_file(
+        config_file, "feast_keyspace", settings["c_keyspace"],
+    )
+    #
+    write_setting_or_remove(
+        config_file,
+        settings["c_protocol_version"],
+        "protocol_version",
+        "c_protocol_version",
+    )
+    # it is assumed that if there's local_dc also there's l.b.p.
+    if settings["c_local_dc"] is not None:
+        replace_str_in_file(
+            config_file, "c_local_dc", settings["c_local_dc"],
+        )
+        replace_str_in_file(
+            config_file, "c_load_balancing_policy", settings["c_load_balancing_policy"],
+        )
+    else:
+        remove_lines_from_file(config_file, "load_balancing:")
+        remove_lines_from_file(config_file, "local_dc:")
+        remove_lines_from_file(config_file, "load_balancing_policy:")
+
+
+def bootstrap():
+    """
+    Bootstrap() will automatically be called
+    from the init_repo() during `feast init`.
+    """
+    from feast.driver_test_data import create_driver_hourly_stats_df
+
+    repo_path = pathlib.Path(__file__).parent.absolute()
+    config_file = repo_path / "feature_store.yaml"
+
+    data_path = repo_path / "data"
+    data_path.mkdir(exist_ok=True)
+
+    end_date = datetime.now().replace(microsecond=0, second=0, minute=0)
+    start_date = end_date - timedelta(days=15)
+    #
+    driver_entities = [1001, 1002, 1003, 1004, 1005]
+    driver_df = create_driver_hourly_stats_df(driver_entities, start_date, end_date,)
+    #
+    driver_stats_path = data_path / "driver_stats.parquet"
+    driver_df.to_parquet(path=str(driver_stats_path), allow_truncated_timestamps=True)
+
+    # example.py
+    example_py_file = repo_path / "example.py"
+    replace_str_in_file(example_py_file, "%PARQUET_PATH%", str(driver_stats_path))
+
+    # store config yaml, interact with user and then customize file:
+    settings = collect_cassandra_store_settings()
+    apply_cassandra_store_settings(config_file, settings)
+
+
+def replace_str_in_file(file_path, match_str, sub_str):
```

Comment:
can you move these to utils so we can use them for other use cases too?

Reply:
done (this led me to create a top-level "file_utils.py' module and touch several template-related modules as well).